### PR TITLE
Feature/website settings

### PR DESCRIPTION
**Pull request by samuelgrant**
Added some backend to dynamically build backends

### CODE REVIEW

File: Resources/js/React/Components/FormControl.js [hunk 1]
```diff
@@ -73,6 +73,8 @@ export class Input extends Component {
                 return "image";
             case "password":
                 return "password";
+            case "url":
+                return "url";
             default:
```
Comment (jacksos101) [inline Resources/js/React/Components/FormControl.js]: Well, I tried to make it better...

```
var conversions = new Map([
    ["date", "datetime-local"],
  ]);

var valid =  [
  "email", 
  "file", 
  "image", 
  "password", 
  "url", 
  "text", 
  "datetime-local"];

var type = conversions.includes(this.props.type) 
  ? conversions.get(this.props.type) : this.props.type;

switch(conversions.includes(type) ? conversions[type] : type){
    case valid.includes(type):
        return type;
    default:
        throw 'You must provide a valid input type';
}
```

But I think I just ended up shifting stuff outside of the switch...

Comment (samuelgrant) [reply Resources/js/React/Components/FormControl.js]: Sorry, i'm confused as to what's going on here.

File: Controllers/AdminPortal/Web/WebSettingsController.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                settings.QLinksTitleB = request.Str("titleB");
+
+                await _Db.SaveChangesAsync();
+                _Logger.LogInformation("{0} updated the website quick link titles", User.AccountName());
+                return Ok();
+            }
+            catch (Exception ex)
+            {
+                _Logger.LogError("Error updating website quick link titles: {0}", ex.Message);
+                _Logger.LogError(ex.StackTrace);
```
[the remark this answers is not in the document]
Comment (samuelgrant) [reply Controllers/AdminPortal/Web/WebSettingsController.cs]: Hey I copied your code from another controller :P

[the remark this answers is not in the document]
Comment (samuelgrant) [reply Controllers/AdminPortal/Web/WebSettingsController.cs]: @jacksos101 Check out issue #12

File: Migrations/WebsiteDataContextModelSnapshot.cs [hunk 3]
```diff
@@ -73,6 +73,29 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.ToTable("PasswordReset");
                 });
 
+            modelBuilder.Entity("Deepcove_Trust_Website.Models.WebsiteSettings", b =>
+                {
+                    b.Property<int>("Id")
+                        .ValueGeneratedOnAdd()
+                        .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);
+
+                    b.Property<string>("ContactEmail");
```
[the remark this answers is not in the document]
Comment (samuelgrant) [reply Migrations/WebsiteDataContextModelSnapshot.cs]: **This is an auto generated class.**
- What exactly do you want me to change?
- Why do you want me to change the output of the auto gen class?

File: Resources/js/React/Components/FormControl.js (hunk 1 above)
Comment (samuelgrant) [reply Resources/js/React/Components/FormControl.js]: You know what. I'm inspired. I will try to improve this function, at a later stage.

Comment (samuelgrant) [reply Resources/js/React/Components/FormControl.js]: #13

**Pull request merged**